### PR TITLE
[CU-8688ytgck] Adds creator compensation analytics and generation cost breakdown

### PR DESCRIPTION
**Pull request by manuelurenah**
image

image

### CODE REVIEW

**Comment by JustMaier** [comment]
Task linked: [CU-8688ytgck Add Creator Compensation](https://app.clickup.com/t/8688ytgck)

File: src/components/Buzz/Rewards/DailyBuzzPayout.tsx [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+// get date options as month from start of year to now
+const dateOptions = monthsUntilNow.reverse().map((month) => {
+  const date = dayjs(month);
+  return {
+    value: date.toISOString(),
+    label: date.format('MMMM YYYY'),
+  };
+});
+
+export const DailyBuzzPayout = () => {
```
Comment (lrojas94) [inline src/components/Buzz/Rewards/DailyBuzzPayout.tsx:58]: I guarantee this name will  get confusing in the future, let's call this component "DailyCreatorCompReward" or something of the sort.

File: src/components/ImageGeneration/GenerationForm/GenerationForm2.tsx [hunk 2]
```diff
@@ -204,7 +205,16 @@ export function GenerationFormContent() {
...
 
-    const { model, resources: additionalResources, vae, remix, aspectRatio, ...params } = data;
+    const {
+      model,
+      resources: additionalResources,
+      vae,
+      remix,
+      aspectRatio,
+      civitaiTip = 0,
+      creatorTip = 0,
```
Comment (lrojas94) [inline src/components/ImageGeneration/GenerationForm/GenerationForm2.tsx:215]: I believe the plan is to default this to 25%

File: src/pages/user/earn-potential.tsx [hunk 3]
```diff
@@ -3,6 +3,7 @@ import { Currency } from '@prisma/client';
 import { IconBolt, IconCategory, IconPercentage } from '@tabler/icons-react';
 import { useRouter } from 'next/router';
 import { useState } from 'react';
+import { DailyBuzzPayout } from '~/components/Buzz/Rewards/DailyBuzzPayout';
```
Comment (lrojas94) [inline src/pages/user/earn-potential.tsx:6]:
```suggestion
```

File: src/server/controllers/buzz.controller.ts [hunk 4]
```diff
@@ -378,3 +380,20 @@ export const claimDailyBoostRewardHandler = async ({ ctx }: { ctx: DeepNonNullab
...
+  ctx,
+}: {
+  input: GetDailyBuzzCompensationInput;
+  ctx: DeepNonNullable<Context>;
+}) {
+  if (!ctx.user.isModerator) input.userId = ctx.user.id;
+  if (!input.userId) input.userId = ctx.user.id;
+
+  try {
+    return getDailyBuzzPayoutByUser({ userId: ctx.user.id, ...input });
```
Comment (lrojas94) [inline src/server/controllers/buzz.controller.ts:395]: Overall same comment with the naming here. We'll likely have Multiple daily buzz payouts as time go by. (my guess/expectation) - so we might wanna clear out that this is genreation related, or creator compensation. Something like that

File: src/server/jobs/deliver-creator-compensation.ts [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+import { withRetries } from '~/server/utils/errorHandling';
+import dayjs from 'dayjs';
+import { TransactionType } from '~/server/schema/buzz.schema';
+import { formatDate } from '~/utils/date-helpers';
+import { createBuzzTransactionMany } from '~/server/services/buzz.service';
+import data from '~/pages/api/generation/data';
+
+export const updateCreatorResourceCompensation = createJob(
+  'update-creator-resource-compensation',
+  '0 * * * *',
```
Comment (lrojas94) [inline src/server/jobs/deliver-creator-compensation.ts:15]: We sure we want this every hour? Sounds overkill

File: src/server/jobs/deliver-creator-compensation.ts [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+  async () => {
+    if (!clickhouse) return;
+
+    const [lastUpdate, setLastUpdate] = await getJobDate(
+      'run-daily-compensation-payout',
+      new Date()
+    );
+    const date = dayjs(lastUpdate).utc().subtract(1, 'day').startOf('day');
+
+    console.log({ date: date.toISOString() });
```
Comment (lrojas94) [inline src/server/jobs/deliver-creator-compensation.ts:97]:
```suggestion
```

File: src/server/jobs/deliver-creator-compensation.ts [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+    }
+
+    const tipBatches = chunk(tipTransactions, BATCH_SIZE);
+    for (const batch of tipBatches) {
+      await withRetries(() => createBuzzTransactionMany(batch), 1);
+    }
+
+    await setLastUpdate();
+
+    console.dir({ creatorsToPay, compensationTransactions, tipTransactions }, { depth: null });
```
Comment (lrojas94) [inline src/server/jobs/deliver-creator-compensation.ts:188]:
```suggestion
```

**Review by lrojas94 — APPROVED** [review]

**Pull request merged**